File: backend/src/main/java/com/manageit/manageit/feature/notification/dto/NotificationDto.java
package com.manageit.manageit.feature.notification.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.manageit.manageit.feature.user.dto.BasicUserDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class NotificationDto {
    private UUID id;
    private BasicUserDto user;
    private String message;
    private LocalDateTime date;
    private UUID projectId;
    private UUID taskId;
}

File: frontend/src/app/features/dto/notification.dto.ts
import { UserSummaryDto } from '@/app/shared/dto/user-summary.dto';

export interface NotificationDto {
  readonly id: string;
  readonly user: UserSummaryDto;
  readonly message: string;
  readonly date: string;
  readonly projectId: string;
  readonly taskId?: string;
}
